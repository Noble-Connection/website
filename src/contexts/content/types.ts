export interface IContentContext {
  charityName: string;
  links: {
    homePage: string;
    aboutUs: string;
    contactUs: string;
    donate: string;
    facebook: string;
    twitter: string;
    youtube: string;
  };
  footer: {
    copyright: string;
    contactNumber: string;
    charityRegistrationDetails: string;
  }
}
