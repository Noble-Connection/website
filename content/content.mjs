export const content = {
  charityName: 'Noble Connection',
  contactNumber: '020 3633 8996',
  emailAddress: 'info@nobleconnection.co.uk',
  navigation: [
    {
      label: 'About Us',
      href: '/about-us',
    },
    {
      label: 'Contact Us',
      href: '/contact-us',
    },
    {
      label: 'Donate',
      href: '/donate',
    },
  ],
  links: {
    facebook: 'https://www.facebook.com/noble.connection/',
    twitter: 'https://twitter.com/nobleconnection?lang=en',
    youtube:
      'https://www.youtube.com/channel/UCzVF8TF4o8wWSX3uNgSheFw/featured',
  },
  footer: {
    copyright: 'Copyright Â©',
    charityRegistrationDetails: 'Charity Reg. No. 1186037',
    address: {
      registeredOfficeText: 'Registered Office',
      firstLine: '102 Islip Manor Road',
      town: 'Northolt',
      postcode: 'UB5 5EB',
      country: 'United Kingdom',
    },
  },
};
