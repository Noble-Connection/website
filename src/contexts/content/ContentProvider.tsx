import { ContentContext } from './ContentContext';
import { IContentContext } from './types';

export const ContentProvider: (params: {
  children: JSX.Element;
  content: IContentContext;
}) => JSX.Element = ({ content, children }) => (
  <ContentContext.Provider value={content}>{children}</ContentContext.Provider>
);
