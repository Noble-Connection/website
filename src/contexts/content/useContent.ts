import { useContext } from 'react';
import { ContentContext } from './ContentContext';
import { IContentContext } from './types';

export const useContent = (): IContentContext =>
  useContext<IContentContext>(ContentContext);
