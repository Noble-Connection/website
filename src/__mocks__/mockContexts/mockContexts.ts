import { IContentContext } from '../../contexts/content/types';

export const mockContentContext: IContentContext = {
  charityName: 'Noble Connection',
  contactNumber: '020 3633 8996',
  emailAddress: 'info@nobleconnection.co.uk',
  pages: {
    aboutUs: 'About Us',
    contactUs: 'Contact Us',
    donate: 'Donate',
  },
  links: {
    homePage: '/',
    aboutUs: '/about-us',
    contactUs: '/contact-us',
    donate: '/donate',
    facebook: 'https://www.facebook.com/noble.connection/',
    twitter: 'https://twitter.com/nobleconnection?lang=en',
    youtube:
      'https://www.youtube.com/channel/UCzVF8TF4o8wWSX3uNgSheFw/featured',
  },
  footer: {
    copyright: 'Copyright Â©',
    charityRegistrationDetails: 'Charity Reg. No. 1186037',
    address: {
      registeredOfficeText: 'Registered Office',
      firstLine: '102 Islip Manor Road',
      town: 'Northolt',
      postcode: 'UB5 5EB',
      country: 'United Kingdom',
    },
  },
};
