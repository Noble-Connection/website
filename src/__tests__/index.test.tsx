import { render, RenderResult } from '@testing-library/react';
import { ChakraProvider } from '@chakra-ui/provider';
import Home from '../pages/index';
import { ContentProvider } from '../contexts/content/ContentProvider';
import { mockContentContext } from '../__mocks__/mockContexts/mockContexts';

describe('Home page', () => {
  const renderHomePage = (): RenderResult =>
    render(
      <ChakraProvider>
        <ContentProvider content={mockContentContext}>
          <Home />
        </ContentProvider>
      </ChakraProvider>,
    );

  it('should render page unchanged', () => {
    const { container } = renderHomePage();

    expect(container).toMatchSnapshot();
  });
});
