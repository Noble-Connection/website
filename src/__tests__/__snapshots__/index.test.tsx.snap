// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders page unchanged 1`] = `
<div>
  <div
    class="container"
  >
    <main
      class="main"
    >
      <h1
        class="title"
      >
        Welcome to 
        <a
          href="https://nextjs.org"
        >
          Next.js!
        </a>
      </h1>
      <p
        class="description"
      >
        Get started by editing
         
        <code
          class="code"
        >
          pages/home.tsx
        </code>
      </p>
      <div
        class="grid"
      >
        <a
          class="card"
          href="https://nextjs.org/docs"
        >
          <h2>
            Documentation →
          </h2>
          <p>
            Find in-depth information about Next.js features and API.
          </p>
        </a>
        <a
          class="card"
          href="https://nextjs.org/learn"
        >
          <h2>
            Learn →
          </h2>
          <p>
            Learn about Next.js in an interactive course with quizzes!
          </p>
        </a>
        <a
          class="card"
          href="https://github.com/vercel/next.js/tree/canary/examples"
        >
          <h2>
            Examples →
          </h2>
          <p>
            Discover and deploy boilerplate example Next.js projects.
          </p>
        </a>
        <a
          class="card"
          href="https://vercel.com/new?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        >
          <h2>
            Deploy →
          </h2>
          <p>
            Instantly deploy your Next.js site to a public URL with Vercel.
          </p>
        </a>
      </div>
    </main>
    <footer
      class="footerContainer"
    >
      <div
        class="css-k008qs"
      >
        <a
          href="https://www.facebook.com/noble.connection/"
        >
          <svg
            fill="currentColor"
            height="21"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="21"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
            />
          </svg>
        </a>
        <a
          class="paddingLeft"
          href="https://twitter.com/nobleconnection?lang=en"
        >
          <svg
            fill="currentColor"
            height="21"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="21"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
            />
          </svg>
        </a>
        <a
          class="paddingLeft"
          href="https://www.youtube.com/channel/UCzVF8TF4o8wWSX3uNgSheFw/featured"
        >
          <svg
            fill="currentColor"
            height="21"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="21"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M549.655 124.083c-6.281-23.65-24.787-42.276-48.284-48.597C458.781 64 288 64 288 64S117.22 64 74.629 75.486c-23.497 6.322-42.003 24.947-48.284 48.597-11.412 42.867-11.412 132.305-11.412 132.305s0 89.438 11.412 132.305c6.281 23.65 24.787 41.5 48.284 47.821C117.22 448 288 448 288 448s170.78 0 213.371-11.486c23.497-6.321 42.003-24.171 48.284-47.821 11.412-42.867 11.412-132.305 11.412-132.305s0-89.438-11.412-132.305zm-317.51 213.508V175.185l142.739 81.205-142.739 81.201z"
            />
          </svg>
        </a>
      </div>
      <div
        class="footerPageLinksContainer"
      >
        <a
          class="chakra-link css-1t4tsbs"
          href="/about-us"
        >
          About Us
        </a>
        <a
          class="chakra-link css-1t4tsbs"
          href="/contact-us"
        >
          Contact Us
        </a>
        <a
          class="chakra-link css-1t4tsbs"
          href="/donate"
        >
          Donate
        </a>
      </div>
      <div
        class="footerContactDetails"
      >
        <address
          class="address"
        >
          <p
            class="chakra-text registeredAddress css-0"
          >
            Registered Office
          </p>
          <p
            class="chakra-text css-0"
          >
            102 Islip Manor Road
          </p>
          <p
            class="chakra-text css-0"
          >
            Northolt
          </p>
          <p
            class="chakra-text css-0"
          >
            UB5 5EB
          </p>
          <p
            class="chakra-text css-0"
          >
            United Kingdom
          </p>
        </address>
        <a
          class="chakra-link css-1t4tsbs"
          href="tel:+4402036338996"
        >
          020 3633 8996
        </a>
        <a
          class="chakra-link css-1t4tsbs"
          href="mailto:info@nobleconnection.co.uk"
        >
          info@nobleconnection.co.uk
        </a>
      </div>
      <div>
        <p
          class="chakra-text css-0"
        >
          Charity Reg. No. 1186037
        </p>
        <br />
        <p
          class="chakra-text css-0"
        >
          Copyright © 2022 Noble Connection
        </p>
      </div>
    </footer>
  </div>
</div>
`;
